core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -762,6 +762,7 @@ private void loadMetadata(LockedInodePath inodePath, LoadMetadataContext context
             LOG.debug("fetching children for {} returned null", inodePath.getUri());
             return;
           }
+          int failedSync = 0;
```

Comment:
The failed count is also counted in `sync()`, how does counting work if we are also counting here?

Reply:
The failed count in `sync()` is the counter of loading request path failed. `failedSync` is the counter of loading children path of request path failed.

Reply:
Thx for the explanation.

It can make the log more confusing if another count is logged without explanation. At least the exception or log should contain which path it is from.

And it can be worse. If I understand correctly, if you have a structure
```
/a
/a/b1
/a/b2
/a/b3
/a/b/c1
/a/b/c2
```
The `loadMetadata()` will be called on `/a` and recursively on `/a/b/`. If some files fail to load, you see one line for `/a/b/` and `/a/` like 
```
2 files failed to load in /a/b
1 files failed to load in /a/
```
It'll be very hard to make sense out of these log messages, because every dir counts for its own.

(No need to implement in this PR) @yuzhu Do you think it makes sense to make `loadMetadata()` return a status object and only log in the top level?

Reply:
i agree it makes sense to log at the top level, but returning a status object might not be the right way because we also want to propagate exception messages up. anyways we can address this in a future PR.

Reply:
The status object can include a list of exceptions (or error objects that only contains the exception class and message, to save some space). An alternative is to use `Exception.addSuppressed(Exception cause)` to avoid losing the causer exceptions.